compiler/ballerina-compiler-api/src/main/java/org/ballerina/compiler/impl/BallerinaSemanticModel.java

```diff
@@ -99,4 +115,16 @@ public BallerinaSemanticModel(BLangCompilationUnit compilationUnit, BLangPackage
     public List<Diagnostic> diagnostics(TextRange range) {
         return new ArrayList<>();
     }
+
+    private boolean isSymbolInUserProject(BSymbol symbol, DiagnosticPos cursorPos) {
+        return symbol.origin == SOURCE &&
+                (cursorPos.compareTo(symbol.pos) > SYMBOL_POSITION ||
```

Comment:
Apparently `SYMBOL_POSITION` constant is redundant and the name is misleading here. Shall we just use `0` for the clarity.

Reply:
Yeah, I agree the name doesn't reflect the actual position of the symbol. Thought of introducing a constant instead of using `0` since the use of `0` kinda seemed to make it a bit unclear what that comparison is about

Reply:
This basically compares whether the cursor position is before or after the position of a given symbol

Reply:
Oh okay. So this is an object compare.
I also feel `0` is more natural in that case.